conn.go

```diff
@@ -885,226 +893,214 @@ func (c *Conn) readLoop(name string, pktConn ipPacketConn, inboundBufferSize int
 		}
 
 		func() {
-			header, err := p.Start(b[:n])
+			var msg dnsmessage.Message
+			err := msg.Unpack(b[:n])
```

Comment:
Since Questions are echoed with Answers you have to at least parse the Answer blocks before knowing how to handle the Questions.